app/scripts/services/events_service.js

```diff
@@ -5,27 +5,48 @@
     .module('app.services')
     .service('EventsService', EventsService);
 
-  EventsService.$inject = ['DataFactory', '$firebaseArray'];
+  EventsService.$inject = ['DataFactory'];
 
-  function EventsService(DataFactory, $firebaseArray) {
+  function EventsService(DataFactory) {
     this.create = create;
 
     function create(create_event) {
       var eventRef = DataFactory('events');
-      $firebaseArray(eventRef);
-      return eventRef.orderByChild("timeTo").equalTo(create_event.timeTo).once('value').then(function(snapshot) {
-        if(snapshot.val() !== null){
-          var keys = Object.keys(snapshot.val());
-          var data = snapshot.val()[keys[0]];
-          if(data.roomName === create_event.roomName && data.timeFrom === create_event.timeFrom) {
-            var error = 'event existed!';
+      var event_conflict =[];
+      eventRef.orderByChild("xxxxx").startAt(create_event.timeFrom+1).endAt(create_event.timeTo).once('value').then(function(snapshot) {
```

Comment:
Please give a `NOTE` for this `xxxxx` attribute that can help other ppl understand your idea.